lib/hexpm/accounts/web_auth_request.ex

```diff
@@ -41,15 +39,16 @@ defmodule Hexpm.Accounts.WebAuthRequest do
 
     audit = {user, request.audit}
 
-    scope = request.scope
-    device_code = request.device_code
-    name = "Web Auth #{device_code} key"
+    key_name = request.key_name
 
-    key_params = %{@key_params | name: name}
-    key_params = %{key_params | permissions: [%{@key_permission | resource: scope}]}
+    key_params = %{@key_params | name: key_name}
+
+    write_key = %{key_params | permissions: [%{@key_permission | resource: "write"}]}
+    read_key = %{key_params | permissions: [%{@key_permission | resource: "read"}]}
```

Comment:
What do you think about letting the user specifying what keys and with what permissions to generate? That way it's more flexible for the future.

Reply:
> What do you think about letting the user specifying what keys and with what permissions to generate? That way it's more flexible for the future.

I think we can always work on that later. Right now, I think we need to get a minimum, viable version ready, which I think this current implementation would be once I figure out the submission/verification form. This would allow me to get a implementation ready for the client without wasting much time, rather than working on more features on the server side.

Reply:
However, I do like the idea of specifying permissions.